src/amo/components/OverallRating.js

```diff
@@ -1,37 +1,69 @@
 import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
 
+import { setUserRating } from 'amo/actions/ratingActions';
+import { callApi } from 'core/api';
 import translate from 'core/i18n/translate';
+import log from 'core/logger';
 
 import 'amo/css/OverallRating.scss';
 
 
 export class OverallRatingBase extends React.Component {
   static propTypes = {
-    i18n: PropTypes.object,
-    addonName: PropTypes.string,
+    addonName: PropTypes.string.isRequired,
+    addonID: PropTypes.number.isRequired,
+    authToken: PropTypes.string,
+    version: PropTypes.object.isRequired,
+    i18n: PropTypes.object.isRequired,
+    createRating: PropTypes.func.isRequired,
+  }
+
+  onChange = ({ currentTarget: form }) => {
+    const input = form.querySelector('input:checked');
+    if (!input) {
+      // I don't think any user input would cause a form change like
+      // this. I'm not sure what would cause this at all but it's probably
+      // better not to throw an exception.
+      log.error('The form changed but no input was selected');
```

Comment:
Perhaps we could group the buttons in a way that shows that they are related? It looks like the `prompt` is currently outside of the form so we could integrate that too.

```html
<form ...>
  <fieldset>
    {/* How do you feel about Ad Blocker? */}
    <legend>{prompt}</legend>
    <label for="love-it">
      <button name="love-it">I love it!</button>
    </label>
    <label for="huh">
      <button name="huh">Huh?</button>
    </label>
    <label for="confused">
      <button name="confused">I'm confused...</button>
    </label>
  </fieldset>
</form>